<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>The bMIND algorithm to estimate sample-level cell-type-specific expression — bMIND • MIND</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="The bMIND algorithm to estimate sample-level cell-type-specific expression — bMIND"><meta property="og:description" content="It calculates the Bayesian estimates of sample- and cell-type-specific (CTS) gene expression, via MCMC. For all input, dimension names are recommended if applicable."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MIND</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/randel/MIND/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>The bMIND algorithm to estimate sample-level cell-type-specific expression</h1>
    <small class="dont-index">Source: <a href="https://github.com/randel/MIND/blob/HEAD/R/bmind_func.r" class="external-link"><code>R/bmind_func.r</code></a></small>
    <div class="hidden name"><code>bMIND.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>It calculates the Bayesian estimates of sample- and cell-type-specific (CTS) gene expression, via MCMC. For all input, dimension names are recommended if applicable.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">bMIND</span><span class="op">(</span>
  <span class="va">bulk</span>,
  frac <span class="op">=</span> <span class="cn">NULL</span>,
  sample_id <span class="op">=</span> <span class="cn">NULL</span>,
  ncore <span class="op">=</span> <span class="cn">NULL</span>,
  profile <span class="op">=</span> <span class="cn">NULL</span>,
  covariance <span class="op">=</span> <span class="cn">NULL</span>,
  nu <span class="op">=</span> <span class="fl">50</span>,
  V_fe <span class="op">=</span> <span class="cn">NULL</span>,
  nitt <span class="op">=</span> <span class="fl">1300</span>,
  burnin <span class="op">=</span> <span class="fl">300</span>,
  thin <span class="op">=</span> <span class="fl">1</span>,
  frac_method <span class="op">=</span> <span class="cn">NULL</span>,
  sc_count <span class="op">=</span> <span class="cn">NULL</span>,
  sc_meta <span class="op">=</span> <span class="cn">NULL</span>,
  signature <span class="op">=</span> <span class="cn">NULL</span>,
  signature_case <span class="op">=</span> <span class="cn">NULL</span>,
  case_bulk <span class="op">=</span> <span class="cn">NULL</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>bulk</dt>
<dd><p>bulk gene expression (gene x sample). We recommend log2-transformed data for better performance, except when using Bisque to estimate 
cell type fractions, raw count is expected for Bisque. If the max(bulk) &gt; 50, bulk will be transformed to log2(count per million + 1) 
before running bMIND.</p></dd>
<dt>frac</dt>
<dd><p>sample-specific cell type fraction (sample x cell type). If not specified (NULL), it will be estimated by non-negative least squares (NNLS) by 
providing signature matrix or Bisque by providing single-cell reference.</p></dd>
<dt>sample_id</dt>
<dd><p>sample/subject ID vector. The default is that sample ID will be automatically provided for sample-level bMIND analysis, otherwise 
subject ID should be provided for subject-level bMIND analysis. Note that the subject ID will be sorted in the output and different sample_id would 
produce slightly different results in MCMCglmm.</p></dd>
<dt>ncore</dt>
<dd><p>number of cores to run in parallel for providing sample/subject-level CTS estimates. The default is all available cores.</p></dd>
<dt>profile</dt>
<dd><p>prior profile matrix (gene by cell type). Gene names should be in the same order of bulk, and cell type names should be in the same order
as frac. If not specified (NULL), the bulk mean will be supplied.</p></dd>
<dt>covariance</dt>
<dd><p>prior covariance array (gene by cell type by cell type). Gene names should be in the same order of bulk, and cell type names should be 
in the same order as frac. If not specified (NULL), bulk variance / sum(colMeans(frac)^2) will be supplied.</p></dd>
<dt>nu</dt>
<dd><p>hyper-parameter for the prior covariance matrix. The larger the nu, the higher the certainty about the information in covariance, and the more 
informative is the distribution. The default is 50.</p></dd>
<dt>V_fe</dt>
<dd><p>hyper-parameter for the covariance matrix of fixed-effects. The default is 0.5 * Identity matrix.</p></dd>
<dt>nitt</dt>
<dd><p>number of MCMC iterations.</p></dd>
<dt>burnin</dt>
<dd><p>burn-in iterations for MCMC.</p></dd>
<dt>thin</dt>
<dd><p>thinning interval for MCMC.</p></dd>
<dt>frac_method</dt>
<dd><p>method to be used for estimating cell type fractions, either 'NNLS' or 'Bisque'. 
**All arguments starting from this one will be used to estimate cell-type fractions only, if those fractions are not pre-estimated.**</p></dd>
<dt>sc_count</dt>
<dd><p>sc/snRNA-seq raw count as reference for Bisque to estimate cell type fractions.</p></dd>
<dt>sc_meta</dt>
<dd><p>meta data frame for sc/snRNA-seq reference. A binary (0-1) column of 'case' is expected to indicate case/control status.</p></dd>
<dt>signature</dt>
<dd><p>signature matrix for NNLS to estimate cell type fractions. Log2 transformation is recommended.</p></dd>
<dt>signature_case</dt>
<dd><p>signature matrix from case samples for NNLS to estimate cell type fractions. Log2 transformation is recommended. If this is 
provided, signature will be treated as signature matrix for unaffected controls.</p></dd>
<dt>case_bulk</dt>
<dd><p>case/control status vector for bulk data when using case/control reference to estimate the cell type fractions for case/control subjects
separately.</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A list containing the output of the bMIND algorithm (some genes with error message in MCMCglmm will not be outputted, 
e.g., those genes with constant expression)</p>
<dl><dt>A</dt>
<dd><p>the deconvolved cell-type-specific gene expression (gene x cell type x sample).</p></dd>
<dt>SE</dt>
<dd><p>the standard error of cell-type-specific gene expression (gene x cell type x sample).</p></dd>
<dt>Sigma_c</dt>
<dd><p>the covariance matrix for the deconvolved cell-type-specific expression (gene x cell type x cell type).</p></dd>
<dt>mu</dt>
<dd><p>the estimated profile matrix (gene x cell type).</p></dd>
<dt>frac</dt>
<dd><p>the estimated cell type fractions (sample x cell type) if fractions are not provided.</p></dd>
</dl></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">example</span><span class="op">)</span></span>
<span class="r-in"><span class="va">bulk</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html" class="external-link">na.omit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">example</span><span class="op">$</span><span class="va">X</span>, <span class="fl">1</span>, <span class="va">as.vector</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in"><span class="va">frac</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html" class="external-link">na.omit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">example</span><span class="op">$</span><span class="va">W</span>, <span class="fl">3</span>, <span class="va">as.vector</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">bulk</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">frac</span><span class="op">)</span> <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">frac</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">bulk</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              1        2       3        4        5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ANK2  4.698981 4.703759 4.88463 5.192639 4.820824</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> POGZ  4.296181 4.502559 4.42363 4.337839 4.102224</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> TRIO  3.143821 3.007979 2.98249 3.508639 2.954794</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> AKAP9 2.648231 2.303789 2.91275 2.900519 1.741124</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ASH1L 3.738681 3.646859 3.75403 4.000939 3.312794</span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">frac</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    astrocytes mature neurons immature neurons oligodendrocytes       OPC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 0.000000000     0.07947864        0.6593213      0.000000000 0.2612001</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 0.004001667     0.17504983        0.6031219      0.022518815 0.1953078</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 0.004323026     0.10218298        0.5877968      0.018830465 0.2868668</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 0.000000000     0.20181933        0.5692321      0.001399302 0.2275493</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 0.000000000     0.18349406        0.5851250      0.026766014 0.2046149</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6 0.005763292     0.07574323        0.7951380      0.000000000 0.1233555</span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># with provided cell type fractions</span></span>
<span class="r-in"><span class="va">deconv1</span> <span class="op">=</span> <span class="fu">bMIND</span><span class="op">(</span><span class="va">bulk</span>, frac <span class="op">=</span> <span class="va">frac</span>, ncore <span class="op">=</span> <span class="fl">12</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># set.seed(1)</span></span>
<span class="r-in"><span class="co"># data(signature)</span></span>
<span class="r-in"><span class="co"># bulk = matrix(rnorm(300 * ncol(bulk), 10), ncol = ncol(bulk))</span></span>
<span class="r-in"><span class="co"># rownames(bulk) = rownames(signature)[1:nrow(bulk)]</span></span>
<span class="r-in"><span class="co"># colnames(bulk) = 1:ncol(bulk)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## without provided cell type fractions: use built-in deconvolution methods NNLS or Bisque to estimate fractions</span></span>
<span class="r-in"><span class="co"># deconv2 = bMIND(bulk, signature = signature[, -6], ncore = 12)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># If you have informative prior, use get_prior() in https://randel.github.io/MIND/reference/get_prior.html to generate prior `profile` and `covariance` matrices. Note that covariance matrix is required to be positive-definite.</span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Jiebiao Wang.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

